subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -733,6 +736,30 @@ static int tls_opt_peer_verify_set(struct net_context *context,
 	return 0;
 }
 
+static int tls_opt_role_set(struct net_context *context, const void *optval,
+			    socklen_t optlen)
+{
+	int *role;
+
+	if (!optval) {
+		return -EFAULT;
```

Comment:
As this is a parameter, shouldn't be better using -EINVAL ?

Reply:
+1. EFAULT means "would lead to MMU fault when accessed", and is usually returned for arbitrary stray pointers. POSIX systems usually don't return EFAULT for NULL pointers, use EINVAL instead.

Reply:
Will fix.